common/rect.h

```diff
@@ -109,91 +143,104 @@ static inline Point operator*(double multiplier, const Point &p) { return Point(
  * When writing code using our Rect class, always keep this principle in mind!
 */
 struct Rect {
-	int16 top, left;		///< The point at the top left of the rectangle (part of the rect).
-	int16 bottom, right;	///< The point at the bottom right of the rectangle (not part of the rect).
+	int16 top, left;		/*!< The point at the top left of the rectangle (part of the Rect). */
+	int16 bottom, right;	/*!< The point at the bottom right of the rectangle (not part of the Rect). */
 
 	Rect() : top(0), left(0), bottom(0), right(0) {}
+	/**
+     * Create a rectangle with width @p w and height @p h.
+     */
 	Rect(int16 w, int16 h) : top(0), left(0), bottom(h), right(w) {}
+	/**
+     * Create a rectangle between the points defined with @p x1, @p y1, @p x2, and @p y2.
+	 * Additionally, check whether it is a valid rectangle.
```

Comment:
The `Additionally, check whether it is a valid rectangle.` is not really correct. It gives the impression that it is safe to pass invalid coordinates. But such a check in a `assert` actually causes a crash if the check fails. It is used to detect incorrect code.

So it would be more correct to say that it create a rectangle with top-left corner at the given x1 and y1 positions, and the bottom-right corner at the given x2 and y2 positions. The x2 value has to be equal or bigger than the x1 value and the y2 value has to be equal or bigger than the y1 value.

Reply:
Changed to 
```
	/**
     * Create a rectangle with top-left corner at the given (x1,y1) position
	 * and the bottom-right corner at the (x2,y2) position.
	 * 
	 * The @p x2 value must be greater or equal @p x1 and @p y2 must be greater or equal @p y1.
     */
```